src/plugins/visualizations/public/react_embeddable/visualize_embeddable.tsx

```diff
@@ -498,6 +498,12 @@ export const getVisualizeEmbeddableFactory: (deps: {
           };
         }, []);
 
+        useEffect(() => {
+          if (hasRendered) {
+            domNode.current?.dispatchEvent(new Event('renderComplete'));
+          }
+        }, [hasRendered]);
```

Comment:
Thanks for opting to support signalling render completion using events. I wonder though if we could leverage the already existing util for doing this in place for constructing the event inline. See [here](https://github.com/elastic/kibana/blob/main/src/plugins/kibana_utils/public/render_complete/render_complete_dispatcher.ts#L13-L15)